OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -207,25 +185,48 @@ void SanityCheckPath(Mobile mobile)
 				throw new InvalidOperationException("(Move) Sanity check failed");
 		}
 
-		Pair<CPos, SubCell>? PopPath(Actor self)
+		// This function could be ported as an extension for Actor
+		bool IsInRange()
+		{
+			var cellRange = nearEnough.Range / 1024;
+			return (mobile.CurrentLocation - destination.Value).LengthSquared <= cellRange * cellRange;
```

Comment:
Calculating range checks in cell coordinates is a bad idea.  I suggest keeping this in WPos / WRange, which has the numerical precision to behave properly for arbitrary diagonals.